std/container/rbtree.d

```diff
@@ -19,6 +19,43 @@ Authors: Steven Schveighoffer, $(WEB erdani.com, Andrei Alexandrescu)
 */
 module std.container.rbtree;
 
+///
+unittest
+{
+    import std.container: RedBlackTree;
+    import std.algorithm: equal;
+
+    auto rbt = new RedBlackTree!(int)(3, 1, 4, 2, 5);
+    assert(rbt.front == 1);
+    assert(equal(rbt[], [1, 2, 3, 4, 5]));
+
+    rbt.removeKey(1, 4);
+    assert(equal(rbt[], [2, 3, 5]));
+
+    rbt.removeFront();
+    assert(equal(rbt[], [3, 5]));
+
+    rbt.insert([1, 2, 4]);
+    assert(equal(rbt[], [1, 2, 3, 4, 5]));
+
+    // Query bounds in O(log(n))
+    assert(rbt.lowerBound(3).equal([1, 2]));
+    assert(rbt.equalRange(3).equal([3]));
+    assert(rbt.upperBound(3).equal([4, 5]));
+
+    // A Red Black tree with the highest element at front:
+    import std.range: iota;
+    auto maxQueue = new RedBlackTree!(int, "a > b")(iota(5));
```

Comment:
Note, the fact that you are using iota means you must use the class ctor. But you could technically do this with the convenience function (I'm surprised it doesn't work with iota):

```D
auto maxQueue = redBlackTree!"a > b"(0, 1, 2, 3, 4);
```